File: Kotlin/Aula-Kotlin-24-08-22/src/main/kotlin/classesabstratas/Gato.kt
package classesabstratas

class Gato(especie: String,
           nomeCientifico: String,
           habitat: String,
           carnivoro: Boolean,
           sexo: String,
           idade: Int,
           val nome: String
) : Animal(especie, nomeCientifico, habitat, carnivoro, sexo, idade) {


    override fun correr() {
        println("Correr como gato")
    }

    override fun dormir() {
        println("Dormir como gato")
    }

    override fun andar() {
        println("Andar como gato")
    }

    override fun comer() {
        println("Comer como gato")
    }

    override fun emitirSom() {
        println("Miar")
    }

    fun ronronar(){
        println("Ronronando como gato")
    }
}

File: JavaScript-TypeScript/Ex-JS-TS-06-09-22/src/gato.ts
import { Animal } from "./animal";

export class Gato extends Animal{




    seMovimentar() {
        console.log("Se movimentar como um gato")
    }


    comer() {
        console.log("Comer como um gato")
    }


    transmitirSom() {
        console.log("Miau Miau")
    }

}